Josh/nix (#382)
earthly/website: blog/_posts/2023-03-14-what-is-nix.md

Yeah, the documentation for Nix is notoriously bad. Even comparatively with a lot of the other functional programming ecosystems. Most of the knowledge you will get either from reading the Nix packages GitHub repository directly or from word of mouth. There isn't really a full tutorial set for next year. I think some people have written ones very, very recently… but they historically haven't existed; you might have to ask somebody or know somebody or be able to read the code. And I just recommend that people just learn how to read the code. That's the only really reliable way you're going to get information about what some of these options are actually doing.

{% include quotes/end.html %}

## Conclusions

### Pure Functions

My main take away after spending some time learning about Nix is that it embraces the functional programming concept of a pure function. If I give a function a certain set of inputs, it will return the same result every time, no matter what. Nix is about building software the same way, whether it's your own software, someone else's software, or your entire OS: You declare all your inputs explicitly and it will be built the same way every time.

> What I learned from talking to Nix users is that what really mattered to them, regardless of how they were using Nix, was its ability to bring the concepts of a pure function to areas of computing where it previously didn't exist.

Reviewer: ```suggestion
What I learned from talking to Nix users is that what really mattered to them, regardless of how they were using Nix, was its ability to bring purely functional programming concepts to computing areas that were previously off-limits. 
```
